fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/KeyExpression.java

```diff
@@ -207,6 +207,18 @@ default boolean hasLosslessNormalization() {
         return !createsDuplicates();
     }
 
+    /**
+     * Method that indicates whether the keys can be stored in a fixed width.  For example, an integer or long can be
+     * stored as a fixed width byte[].
+     *
+     * This is utilized by Lucene to see if we can optimize the key lookup in the case of deletes.
+     *
+     */
+    @API(API.Status.INTERNAL)
+    default boolean isFixedWidth(Descriptors.Descriptor descriptor) {
```

Comment:
I don't know if we ever solved this problem, but there's sort of a tension whenever we add a feature that involves walking over the `KeyExpression` tree between adding one function with `instanceof` checks over the various key expression types versus adding a new expression to the `KeyExpression` interface. We definitely used to favor the latter, until we had too many, and now we've kind of gone towards the former. Given that this is only going to be useful for the `Lucene` index and its return value is very much baked in with the Lucene implementation, I think we probably want to go with a function in the Lucene code that knows how to handle the various key expression types

(Fwiw, I think I also experimented with a proper "visitor" class, with "visitX" methods for each of the various `KeyExpression` types, and it worked okay and gave you proper compiler errors if you forgot a type, though it has problems with extending to user-defined `KeyExpression` types, though we also discourage those, now, as `FunctionKeyExpression` gets you most of what you want.)